### PR TITLE
Backport 2.7: fix  test issues

### PR DESCRIPTION
**Pull request by RonEld**
## Description
Backport of ARMmbed/mbed-crypto#278 to `mbedtls-2.7`

## Status
**READY**

### CODE REVIEW

File: tests/suites/test_suite_pkcs1_v15.function [hunk 1]
```diff
@@ -14,23 +14,22 @@ void pkcs1_rsaes_v15_encrypt( int mod, int radix_N, char *input_N, int radix_E,
...
                                char *result_hex_str, int result )
 {
-    unsigned char message_str[1000];
-    unsigned char output[1000];
-    unsigned char output_str[1000];
-    unsigned char rnd_buf[1000];
+    unsigned char message_str[118];
+    unsigned char output[128];
+    unsigned char output_str[257];
+    unsigned char rnd_buf[109];
```
Comment (Patater) [inline tests/suites/test_suite_pkcs1_v15.function]: These were determined experimentally?

File: tests/suites/test_suite_rsa.function [hunk 2]
```diff
@@ -1374,7 +1362,6 @@ void mbedtls_rsa_import_raw( char *input_N,
 
     if( strlen( input_E ) )
         lenE = unhexify( bufE, input_E );
-
```
Comment (Patater) [inline tests/suites/test_suite_rsa.function]: These style changes are to make these tests closer in style to 2.16 and development, right?

File: tests/suites/test_suite_pkcs1_v15.function (hunk 1 above)
Comment (RonEld) [reply tests/suites/test_suite_pkcs1_v15.function]: yes, I ran the test, and printed the lengths. I put here the maximal lengths used.
This is prone to errors obviously, once new test vectors will arrive

File: tests/suites/test_suite_rsa.function (hunk 2 above)
Comment (RonEld) [reply tests/suites/test_suite_rsa.function]: what style changes?
If you mean the removal of this line, that it was caused when I deleted the `printf` of all the lengths used. Should I revert these line deletions?

Comment (Patater) [reply tests/suites/test_suite_rsa.function]: I don't see any printf in the original diff. Removing the lines adds noise to the diff. It's a bit sloppy if your adding and removing printfs for your own testing purposes results in any diffs.

If the style in 2.16 is not to have newlines at these places in the tests, this is OK.

**Comment by gilles-peskine-arm** [comment]
Some test suites failed on the CI.

**Comment by RonEld** [comment]
@Patater @dgreen-arm 
I have amended the commits with reversion of the line deletions and with updated buffer sizes, that caused the CI to fail.
Please review

**Pull request merged**